Gopkg.toml

```diff
@@ -46,15 +46,31 @@ required = [
 
 [[constraint]]
   name = "sigs.k8s.io/cluster-api"
-  revision = "0f315e945d5fabe0c805ccf30fa6cf02fac6c0b5"
+  revision = "c51a9c13aa0ebe6d2d7bef869094b9b78bebcd48"
 
 [[constraint]]
   name = "github.com/Azure/azure-sdk-for-go"
-  version = "25.1.0"
+  version = "26.2.0"
 
 [[constraint]]
   name = "github.com/Azure/go-autorest"
-  version = "11.4.0"
+  version = "11.5.0"
+
+[[override]]
+  name = "k8s.io/api"
+  version = "kubernetes-1.13.4"
+
+[[override]]
+  name = "k8s.io/apiextensions-apiserver"
+  version = "kubernetes-1.13.4"
+
+[[override]]
+  name = "k8s.io/apimachinery"
+  version = "kubernetes-1.13.4"
+
+[[override]]
+  name = "k8s.io/client-go"
+  version = "kubernetes-1.13.4"
```

Comment:
Are these k8s overrides actually needed? The k8s version should be pulled in from cluster-api which pulls it in from controller-runtime.

Reply:
I don't think we should be using overrides.

Reply:
@vincepri / @tariq1890 -- See here in capi:
- https://github.com/kubernetes-sigs/cluster-api/blob/master/Gopkg.lock#L819
- https://github.com/kubernetes-sigs/cluster-api/blob/master/Gopkg.lock#L833
- https://github.com/kubernetes-sigs/cluster-api/blob/master/Gopkg.lock#L888
- https://github.com/kubernetes-sigs/cluster-api/blob/master/Gopkg.lock#L979

CAPI is pulling 1.13.1. I'd like to maintain consistency with the k8s version bump in this PR (1.13.4), hence the overrides.

If the capi versions get constraints for those packages, then I can plan to remove these overrides later.

Reply:
I would prefer to update CAPI and have that pulled transitively. I don't think we should commit an override dep to master.

Reply:
PR created for this https://github.com/kubernetes-sigs/controller-runtime/pull/349

Let's go with this approach. I've had plenty of dependency nightmares because of overrides.

Reply:
@justaugustus This is a common problem for any provider, but I agree with @tariq1890 to stick with controller-runtime's version for now. That won't stop you from deploying a cluster with an updated version though.

Reply:
SGTM! I've removed the override, set a constraint for client-go @ `kubernetes-1.13.1`, and added a TODO to Gopkg.toml which links this comment thread and the dependency bump Tariq PR-ed here: kubernetes-sigs/controller-runtime#349

@tariq1890 / @soggiest -- If everything looks good, this is ready to be re-`/lgtm`ed. :)